src/plugins/intel_gpu/src/plugin/transformations/convert_matmul_to_fc.cpp

```diff
@@ -0,0 +1,172 @@
+                shape_a_aligned.insert(shape_a_aligned.begin(), 1);
+            }
+            for (size_t i = 0, cnt = max_size - rank_b; i < cnt; ++i) {
+                shape_b_aligned.insert(shape_b_aligned.begin(), 1);
+            }
+
+            if (matmul->get_transpose_a()) {
+                std::swap(*(shape_a_aligned.end() - 1), *(shape_a_aligned.end() - 2));
+            }
+            if (matmul->get_transpose_b()) {
+                std::swap(*(shape_b_aligned.end() - 1), *(shape_b_aligned.end() - 2));
+            }
+
+            // check on per-batch MatMul which can't be converted to FC
+            for (size_t i = 0; i < max_size - 2; ++i) {
+                if (shape_b_aligned[i] == 1) {
+                    shape_b_aligned[i] = shape_a_aligned[i];
+                } else {
+                    return std::make_tuple(false, std::move(shape_a_aligned), std::move(shape_b_aligned));
+                }
+            }
+            return std::make_tuple(true, std::move(shape_a_aligned), std::move(shape_b_aligned));
+        };
+
+        /*
+         *  create_transpose function return Transpose operation to replace transpose_a or transpose_b
+         *  arguments with an operation. In other words in this function we create Transpose operation
+         *  with order length equal to output_shape length of given node and fill order with increasing
+         *  sequence starting from 0 and replace last two dimension. For example for length = 4  the
+         *  order will be [0, 1, 3, 2] that emulates transpose_a or transpose_b attribute.
+         */
+        ov::NodeVector new_ops;
+
+        auto create_transpose = [this, &new_ops ](const ov::Output<ov::Node>& node, const std::string& transpose_name) {
+            std::vector<size_t> transpose_order(node.get_partial_shape().size());
+            std::iota(transpose_order.begin(), transpose_order.end(), 0);
+            std::swap(*(transpose_order.end() - 1), *(transpose_order.end() - 2));
+
+            auto transpose_const = ov::op::v0::Constant::create(ov::element::i32, ov::Shape{ transpose_order.size() }, transpose_order);
+            auto transpose = std::make_shared<ov::op::v1::Transpose>(node, transpose_const);
+            if (!ov::is_type<ov::op::v0::Constant>(transpose)) {
+                new_ops.push_back(transpose_const);
+                MatcherPass::register_new_node(transpose);
+            }
+            transpose->set_friendly_name(transpose_name);
+            ov::disable_constant_folding(transpose);
+            new_ops.push_back(transpose);
+            return transpose;
+        };
+
+        bool success = true;
+        ov::PartialShape shape_a_aligned;
+        ov::PartialShape shape_b_aligned;
+        std::tie(success, shape_a_aligned, shape_b_aligned) = get_aligned_shapes();
+        if (!success) {
+            return false;
+        }
+
+        auto aligned_a_rank = shape_a_aligned.rank();
+        auto aligned_b_rank = shape_a_aligned.rank();
```

Comment:
```suggestion
        auto aligned_b_rank = shape_b_aligned.rank();
```
But I'm not sure we need this check at all, because we already have `OPENVINO_ASSERT(shape_b.is_static());` aboce